src/shared/parameter_v2/config/examples/Example.yaml

```diff
@@ -0,0 +1,152 @@
+    description: >
+      Accepts all registered IndexType values for the GenericFactory<index_type, type_to_create>.
+
+      For example, if there was a GenericFactory<std::string, Play> with 3 entries registered
+      as ("Stop", "Defense", "Offense"), then this parameter would only accept
+      "Stop", "Defense", and "Offense" as valid values.
+
+      Will only generate for CPP code.
+
+#########################################################################################
+# The following are example of special attributes that can be applied to any parameter
+#########################################################################################
+-
+# Make a parameter value optional (CPP only)
+- bool:
+    name: example_optional_parameter
+    value: false
+    optional: true
+    description: >
+      Optional parameters will generate with std::optional as their value return type.
+      They can accept std::nullopt as a value. If optional is not specified, it is
+      assumed to be false.
+
+      Will only generate for CPP code.
+# Make a parameter a constant value
+- int:
+    name: example_const_parameter
+    min: 0
+    max: 255
+    value: 50
+    const: true
+    description: >
+      Parameters with the const attribute do not generate with an API to change their values.
+      They are constants. If const is not specified, it is assumed to be false.
+
+      Only parameters that are marked as const will be generated for C code. We assume
+      that C code is firmware that we cannot change at runtime because it is running
+      on the robots (not the main AI computer).
+
+#########################################################################################
+# The following are examples of how to include other configs
+#########################################################################################
+# Config files can include other configs. This is how the parameter hierarchy is built
+# up, and allows us to share parameters between configs. For example, this makes it
+# easy to define some passing parameters in one location, and share those parameters
+# with the subset of Plays that need them.
+#
+# All that needs to be specified is the name the config will be reference as, and the
+# path to the file to include. The include paths follow the same concepts at Bazel:
+# - Files in the same directory can be specified with a colon preceding the filename
+#   Eg. include: ":SomeFile.yaml"
+# - Otherwise, the absolute filepath relative to the root of the config directory
+#   (ie. the top-level directory containing all the config files) must be provided.
+#   The name of the file to be included must be the last component of the path, and
+#   separated by a colon.
+#   Eg. include: "Config/examples:Example.yaml"
+#
+# In the generated code, included configs and their attributes can be accessed by
+# the name of the included config.
+# Eg. example_included_config_in_the_same_directory.foo_bool
+
+- config:
```

Comment:
I think we should put includes at the top like protobuf, and just have the files as a list
Since we say a file is a config, its redundant to name it again imo
~~~
include:
- foo.yaml
- test.yaml
~~~

Reply:
That does create some trickyness if we have two configs with the same name at different parts of the tree. For example:
```
A
  B.yaml
C
  B.yaml
D
   B.yaml
   E.yaml
```
and say in `E.yaml` we have:
```
include:
- //A:B.yaml
- //C:B.yaml
- :B.yaml
```
Butttttt since we only ever want there to be a single instance of a config, I think it's totally fine to just disallow naming configs the same thing to prevent confusion.

Reply:
How would we enforce disallowing naming configs the same thing?

Reply:
I can throw generation-time errors if there are files in the config tree with the same name.

@garethellis0 are you fine with @akhilveeraghanta 's suggestion then if we disallow duplicate file names? We do lose the ability to "name" the included configs, but that's not a huge deal since config files _should_ be named reasonably (just goes back to our in-person discussion about making it as hard as possible to write something invalid, and giving very clear, local, error messages).

Reply:
In terms of generating `C` structs, would we change the name from `CamelCase` to `snake_case`? Or would we just name the files with `snake_case` and then create the types by changing to `CamelCase`?

Reply:
Ah yea I forgot about different naming conventions for different outputs. I originally wanted to avoid mutating names (eg. `snake_case` to `CamelCase`) during generation because I thought it would be misleading, but given that we have different standards for different outputs I think that's actually fine. It can be part of the spec for each language.

I'll hopefully get example generated code up soon which will help clarify this too.

@garethellis0 please confirm if you're ok with the
```
include:
- foo.yaml
- test.yaml
```

if includes, or would rather keep a version similar to what we have currently that allows includes to be named.

Reply:
Talked to @garethellis0 , he agrees with akhil's suggestion to list includes at the top.